### ISSUE TITLE
View initialization, and viewoptions

### ISSUE THREAD

**Issue by brunoscopelliti** [AmpersandJS/ampersand-view#102]
Hi there,

I would know if there is any particular reason behind the fact that in the ampersand-view constructor, the view option attributes are updated only after the  initialize method is called?

```
this.on('change:el', this._handleElementChange, this);

// ...

this.initialize.apply(this, arguments);
this.set(_.pick(attrs, viewOptions));
```

I think having the options set before the view is initialized is the preferred behaviour; otherwise the same code is run twice, without a real need.

In my particular case, I am extending ampersand-view, with a module that retrieve a template from server (or local storage when available)... I will release it open source when I will reach an adeguate test coverage.

During the initialization i've to change the template (root element, ...) 

when `this.set(_.pick(attrs, viewOptions));` is called `attrs.el` still references the old container, the one that was replaced with the server response... and this causes the fact that the event listeners are delegated to the wrong element.

I would suggest to change the order of:

```
this.set(_.pick(attrs, viewOptions));
this.initialize.apply(this, arguments);
```

Do you see any problem? I would be happy to submit a pull request for this.
Thanks for your time :)

**Comment by brunoscopelliti** [comment]
I forked the repo, and made the "fix"...
it passes all the tests...

I hope this can be merged soon... 
if you need more info, i am available for further clarification.

**Comment by pgilad** [comment]
@wraithgar Backbone does this as well. This call the `initialize` function last.
https://github.com/jashkenas/backbone/blob/master/backbone.js#L1171

Although I suggest this PR should be merged, it is a **breaking change**.

There should also be a test that validates the order.

**Comment by gonsfx** [comment]
Should no longer be needed once `viewOptions` are removed.

**Comment by pgilad** [comment]
@brunoscopelliti Thanks for the fix, but once we merge #125 this will be fixed as well..

**Comment by brunoscopelliti** [comment]
@pgilad you're welcome. thank you.
